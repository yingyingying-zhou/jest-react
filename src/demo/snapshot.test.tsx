import React from "react";
import { render, cleanup } from "@testing-library/react";
import App from "../App";

// afterEach(cleanup); // 在每个测试用例执行之后调用（调用多次），避免内存泄漏
// 现在cleanup 都是自动调用的，所以你已经不再需要再考虑它了

it("should take a snapshot", () => {
  const RenderText = () => {
    return <App >test</App>
  }
  const { asFragment } = render(<RenderText />);//  asFragment

  expect(asFragment(<RenderText />)).toMatchSnapshot();
});
