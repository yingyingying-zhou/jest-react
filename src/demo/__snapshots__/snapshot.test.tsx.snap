// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div>
    <label
      for="toggle"
    >
      Show Message
    </label>
    <input
      id="toggle"
      type="checkbox"
    />
  </div>
</DocumentFragment>
`;
